GoogleAnalyticsProvider.m

```diff
@@ -0,0 +1,180 @@
+    if (!category) {
+        category = @"default"; // category is a required value
+    }
+    // Prepare properties dictionary
+    if (!properties) {
+        properties = @{@"value": @([interval intValue])};
+    } else {
+        NSMutableDictionary *newProperties = [properties mutableCopy];
+        newProperties[@"value"] = @([interval intValue]);
+        properties = newProperties;
+    }
+    
+    // Send event
+    [self event:event withProperties:properties];
+    
+    // By Google's header, the interval should be seconds in milliseconds.
+    GAIDictionaryBuilder *builder = [GAIDictionaryBuilder createTimingWithCategory:category
+                                                                          interval:@((int)([interval doubleValue]*1000))
+                                                                              name:event
+                                                                             label:nil];
+    [self send:[self finalizedPropertyDictionaryFromBuilder:builder withProperties:properties]];
+}
+
+- (NSMutableDictionary *)finalizedPropertyDictionaryFromBuilder:(GAIDictionaryBuilder *)builder
+                                                 withProperties:(NSDictionary *)properties {
+    NSMutableDictionary *finalizedProperties = [builder build];
+
+#ifdef DEBUG
+    [self warnAboutIgnoredProperties:properties];
+#endif
+
+    // adding custom Dimension values if we can find a key in the mappings
+    for (NSString *key in self.customDimensionMappings.allKeys) {
+        NSString *potentialValue = properties[key];
+        if (potentialValue) {
+            [finalizedProperties setObject:potentialValue forKey:self.customDimensionMappings[key]];
+        }
+    }
+
+    // adding custom Metric values if we can find a key in the mappings
+    for (NSString *key in self.customMetricMappings.allKeys) {
+        NSString *potentialValue = properties[key];
+        if (potentialValue) {
+            [finalizedProperties setObject:potentialValue forKey:self.customMetricMappings[key]];
+        }
+    }
+
+    return finalizedProperties;
+}
+
+#pragma mark - Dispatch
+
+- (void)dispatchGA {
+    [[GAI sharedInstance] dispatch];
+}
+
+- (void)send:(NSDictionary *)parameters {
+	[self.tracker send:parameters];
+	//	send events immediately while debugging
+#ifdef DEBUG
+	[self dispatchGA];
+#endif
+}
+
+#pragma mark - Warnings
+
+- (void)warnAboutIgnoredProperties:(NSDictionary*)propertiesDictionary {
+    for (NSString *key in propertiesDictionary) {
+        if ([key isEqualToString:[NSDictionary googleAnalyticsLabelKey]] ||
+            [key isEqualToString:[NSDictionary googleAnalyticsCategoryKey]] ||
+            [key isEqualToString:[NSDictionary googleAnalyticsValueKey]] ||
+            [self.customDimensionMappings.allKeys containsObject:key]) {
+            continue;
+        }
+        NSLog(@"%@: property ignored %@:%@",self,key,propertiesDictionary[key]);
+    }
+}
+
+#endif
+
+@end
```

Comment:
And we already have the file for `GoogleAnalyticsProvider.m` [here: /Providers/GoogleAnalyticsProvider.m](https://github.com/orta/ARAnalytics/blob/master) - can you edit that, rather than creating a new one in the root of the folder please?